docs-new/mkdocs.yml

```diff
@@ -0,0 +1,96 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+site_name: Apache Iceberg
+docs_dir: home
```

Comment:
Is there a more obvious name? I think most people expect `docs` for mkdocs projects but I see why you'd want to change it so we don't have `docs/docs`. Before, we had a top-level `site` folder and left this as `docs`.

Reply:
Yeah, but in mkdocs `site` is reserved for the generated site for each mkdocs project. We can also overwrite this, but I wanted to follow convention as much as possible.

Reply:
Yeah, we'd have site/site. I think that's fine and probably better than `home` because people don't care about generated paths. Where they go looking for docs to change is more important.

Reply:
Well, actually site is generated in the same directory as the mkdocs folder, so we would need to rename the site directory. I actually like what Jekyll does and uses underscores to denote the special directories. 

I'll use the `docs_dir: site` for the static pages and `site_dir: _site` for the generated pages. 

To avoid nesting docs versioned sites by yet another layer, i'll keep `docs_dir: .` for the versioned docs but still use the `site_dir: _site` convention there as well. That will make it obvious that all `_site` directories are generated and to be ignored.

Let me know if you take any issue with this approach.

Reply:
I don't understand what you're proposing, but I think that we should go with the defaults inside the mkdocs area. That is, `docs` should contain the source markdown files and is located in the same folder as the top-level `mkdocs.yaml`. `site` also lives in that folder when is it auto-generated. The folder containing all 3 can be called `site` or `docs`, I don't really have a preference there.

Reply:
This causes warnings with the build expecting to see files in the relative locations across docs sites. I am using the current setup to try and minimize these errors so that it can be a notification that a link has broken.

Further, I'd like to continue moving towards a 1:1 relationship between the directory structure, and the sitemap, so newcomers can use the site layout to guide where they will find the file.